#1194 Portuguese translations

nmadruga opened the pull request:
Fixes #1194 

1. Fixed text displayed at Review History page.
2. Fixed options displayed on filters (in list) for Reviewer and Assigner actions
3. Updated Localisation.md file with instructions for working with localisations
4. And remove one misleading string

CarlosNZ reviewed (changes requested):
Thanks -- just need to move the string mappings into `useLocalisedEnums` and keep the same pattern

nmadruga commented on src/utils/hooks/useGetQuestionReviewHistory.tsx:
```diff
@@ -39,6 +39,13 @@ const useGetQuestionReviewHistory = ({ isApplicant, ...variables }: UseGetQuesti
     ? useGetHistoryForApplicantQuery({ variables })
     : useGetHistoryForReviewerQuery({ variables })
 
+    const localisedResponse: { [key: string]: string} = {
+      'APPROVE': strings.LABEL_REVIEW_APPROVED,
+      'DECLINE': strings.LABEL_REVIEW_DECLINED,
+      'AGREE': strings.LABEL_CONSOLIDATION_AGREEMENT,
+      'DISAGREE': strings.LABEL_CONSOLIDATION_DISAGREEMENT
+    }
```
Sure! I didn't find where it was! Thank you. Maybe I should add this to the new Localisations docs, so it helps for future work done around that : )

CarlosNZ replied:
That localisation doc is currently written from a user perspective, so I don't think we should clutter it up with internal details. Feel free to start an internal "localisation system" doc, although it'd need to have an explanation of how the localisation context works, how plugins get their strings, etc. so not sure you want to do that just yet.

I guess what I'm saying is it'd be weird to document this one very specific thing when there's more fundamental aspects of the localisation system we haven't yet documented.

nmadruga replied:
I have added to the same doc, ideally this is just basic material for dev/admins to use. A User Guide to be done separately for final users - where some of this work can be copied/based from. 

I didn't go into much details of contexts. I think we can have special "Learning" sections on the App to talk about technologies in use, instead of adding as part some specific note about how to add/do management of localisations

nmadruga commented:
Hi! I have done the fixes - and added 2 more enums to be on `useLocalisedEnums`.  Added missing localised strings in 2 plugins: `Number` and `Checkbox`. Have also done additions to the Localisation.md.

CarlosNZ commented on docs/internal/admin/Localisation.md:
```diff
@@ -36,4 +38,47 @@ Workflow for exporting and importing could be as follows:
+
+### How localisation works?
+
+The app always come with the file `defaultLanguageStrings.js` file which has all the keys of localisation and the base English strings to be used - in case there is a gap on the **main** language to be used (missing key) or even missing the language file... To define other languages to be used is usually done by exporting the existing languages and reimporting with a new language and its strings  ...
```
```suggestion
The app comes with a base`defaultLanguageStrings.ts` file which defines all the localisation keys and the base English strings to be used. These are also used as fallbacks if the currently active language is missing some translations.

Defining other languages is usually done by exporting the existing languages and reimporting with the new language's strings (as explained above in **Localisation UI (external users)**). The localisation files are stored in the folder in the server: `./localisation`. They are also exported and imported with [Snapshots](https://github.com/openmsupply/conforma-server/wiki/Snapshots)
```

CarlosNZ commented on docs/internal/admin/Localisation.md:
```diff
@@ -36,4 +38,47 @@ Workflow for exporting and importing could be as follows:
+
+### Localisations directory
+
+ Contains one languages configuration file `languages.json` and each language folders each containing one file `strings.json` with a map of strings (key) and localised text. The configuration file is easy to follow, and relates to the other language folder in the same directory:
```
```suggestion
Contains one overall languages configuration file `languages.json` and separate folders for each language containing one `strings.json` file which consists of a map of strings (keys) and localised text.

The configuration file is easy to follow, and relates to the other language folder in the same directory:
```

CarlosNZ commented on docs/internal/admin/Localisation.md:
````diff
@@ -36,4 +38,47 @@ Workflow for exporting and importing could be as follows:
+]
+ ```
+
+ The configuration file bascially has which languages are selectable by the user (`enabled` option), which how it displays to be user for selecion. In case only one language is available, the user don't the 'Language selection' option on the UI. In other cases it will display as the following:
````
```suggestion
The configuration file specifies which languages are selectable by the user (`enabled` option), along with how it is displayed to a user in the language selection UI.  If only one language is available, the user doesn't see this option. In other cases it will display like this:
```

CarlosNZ commented on src/utils/hooks/useLocalisedEnums.ts:
```diff
@@ -34,7 +47,25 @@ const useLocalisedEnums = () => {
   }
 
-  return { Status, Outcome, Decision }
+  const ReviewAction: { [key in ReviewerAction]: string } = {
+    SELF_ASSIGN: strings.ACTION_SELF_ASSIGN,
+    START_REVIEW: strings.ACTION_START,
+    VIEW_REVIEW: strings.ACTION_VIEW,
+    CONTINUE_REVIEW: strings.ACTION_CONTINUE,
+    RESTART_REVIEW: strings.ACTION_PENDING,
+    MAKE_DECISION: strings.ACTION_MAKE_DECISION,
+    UPDATE_REVIEW: strings.ACTION_UPDATE
+  }
+
+  const ReviewResponse: { [key in ReviewResponseDecision]: string } =
+    {
+      APPROVE: strings.LABEL_REVIEW_APPROVED,
+      DECLINE: strings.LABEL_REVIEW_DECLINED,
+      AGREE: strings.LABEL_CONSOLIDATION_AGREEMENT,
+      DISAGREE: strings.LABEL_CONSOLIDATION_DISAGREEMENT,
+    }
```
Thanks, good to have this list complete now :)

CarlosNZ reviewed (commented):
Cool, thanks -- just made a few suggestions regarding typos and simplifying the grammar a bit.

nmadruga commented on docs/internal/admin/Localisation.md:
````diff
@@ -36,4 +38,47 @@ Workflow for exporting and importing could be as follows:
+
+The app always come with the file `defaultLanguageStrings.js` file which has all the keys of localisation and the base English strings to be used - in case there is a gap on the **main** language to be used (missing key) or even missing the language file... To define other languages to be used is usually done by exporting the existing languages and reimporting with a new language and its strings  ...
+
+### Localisations directory
+
+ Contains one languages configuration file `languages.json` and each language folders each containing one file `strings.json` with a map of strings (key) and localised text. The configuration file is easy to follow, and relates to the other language folder in the same directory:
+ ```
+ [
+  {
+    "languageName": "English",
+    "description": "NZ English",
+    "code": "en_nz",
+    "flag": "🇳🇿",
+    "enabled": true
+  },
+  {
+    "languageName": "Portugese",
+    "description": "Portugese translations",
+    "code": "pt_br",
+    "flag": "🇧🇷",
+    "enabled": false
+  },
+  {
+    "languageName": "English",
+    "description": "Certain string replacements for DTAC scholarship",
+    "code": "en_dtac",
+    "flag": "🇦🇺",
+    "enabled": true
+  }
+]
+ ```
````
I think it's alright, since this is only explaining how/where to define the localisation, but below there is this line already which says this will be loaded on the back-end: 
> The localisation files are stored in the folder in the server:

nmadruga commented:
@CarlosNZ let me know if all good to merge this one now

CarlosNZ commented on docs/internal/admin/Localisation.md:
```diff
@@ -42,15 +42,19 @@ Workflow for exporting and importing could be as follows:
 ### How localisation works?
 
-The app always come with the file `defaultLanguageStrings.js` file which has all the keys of localisation and the base English strings to be used - in case there is a gap on the **main** language to be used (missing key) or even missing the language file... To define other languages to be used is usually done by exporting the existing languages and reimporting with a new language and its strings  ...
+The app comes with a base`defaultLanguageStrings.ts` file which defines all the localisation keys and the base English strings to be used. These are also used as fallbacks if the currently active language is missing some translations.
```
```suggestion
The app comes with a base `defaultLanguageStrings.ts` file which defines all the localisation keys and the base English strings to be used. These are also used as fallbacks if the currently active language is missing some translations.
```
Now a missing space lol

CarlosNZ reviewed (approved):
Cool, thanks, looks good.

Couple of tiny whitespace nits if you just want to deal with them before you merge.
